include/graph_generator.h

```diff
@@ -19,6 +19,14 @@ namespace detail {
 	class abstract_command;
 	class horizon_command;
 
+	// TODO: Move to utility header..?
+	struct pair_hash {
+		template <typename U, typename V>
+		std::size_t operator()(const std::pair<U, V>& p) const {
+			return p.first ^ p.second;
```

Comment:
XOR of 2 values doesn't seem like that good a hash, maybe you intended something else ?

Reply:
Thanks! You are right of course, I meant to combine the two hashes, not the values. I've changed this to use a re-implementation of Boost's `hash_combine`, which seems to be an accepted "good way" of doing it